src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,190 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import core.basesyntax.service.exception.RegisterException;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDaoImpl storageDao;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    void register_userAlreadyExists_notOk() {
+        User user = new User();
+        user.setLogin("ExistsLogin");
+        user.setPassword("ValidPassword");
+        user.setAge(25);
+        storageDao.add(user);
+        assertThrows(RegisterException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void loginIsValid_nullLogin_notOk() {

```

Comment:
You actually test method register, so let's stick to the common mistakes describes the naming 

Remember about naming of test methods. There are a lot of ways to name your test methods. The main point is that they should have informative names and be consistent along with other developers in your team. For this task use such convention: <methodUnderTest>_<state>_<expectedBehavior>. For example, if we are testing the method register with a null user’s age the test method name should be register_nullAge_notOk. notOk is because the test expects the register method to throw an exception.

Check all places